### PR TITLE
Add clover symbol

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Add 'clover' keyword to Pipeline DSL

Adds a 'clover' keyword to the Pipeline domain specific language so that Pipeline Snippet Syntax Generator is able to guide users on the syntax of the clover task.

Also adds a test for scripted Pipeline with the 'clover' keyword and for declarative Pipeline.

### CODE REVIEW

File: src/test/java/hudson/plugins/clover/CloverWorkflowTest.java [hunk 1]
```diff
@@ -12,6 +12,10 @@
 
 public class CloverWorkflowTest {
 
+    // BuildWatcher echoes job output to stderr as it arrives
+    @org.junit.ClassRule
```
Comment (marek-parfianowicz) [inline src/test/java/hudson/plugins/clover/CloverWorkflowTest.java:16]: Use `import` for org.junit.ClassRule and org.jvnet.hudson.test.BuildWatcher, please.

File: src/test/java/hudson/plugins/clover/DeclarativePipelineTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                "            </class>",
+                "            <class name=\"CloverPublisher.DescriptorImpl\">",
+                "               <metrics conditionals=\"0\" methods=\"4\" coveredstatements=\"0\" coveredmethods=\"0\" coveredconditionals=\"0\" statements=\"6\" coveredelements=\"0\" elements=\"10\"/>",
+                "            </class>",
+                "            <metrics conditionals=\"0\" methods=\"8\" classes=\"2\" ncloc=\"42\" coveredstatements=\"0\" coveredmethods=\"0\" coveredconditionals=\"0\" statements=\"11\" loc=\"79\" coveredelements=\"0\" elements=\"19\"/>",
+                "            <line num=\"27\" count=\"0\" type=\"method\"/>",
+                "         </file>",
+                "      </package>",
+                "   </project>",
+                "</coverage>",
```
Comment (marek-parfianowicz) [inline src/test/java/hudson/plugins/clover/DeclarativePipelineTest.java:47]: I recommend moving this to an external file, put it in `src/test/resources` directory.

We could have a helper function to read this resource file into String and next concatenate with the rest.

Comment (marek-parfianowicz) [reply src/test/java/hudson/plugins/clover/DeclarativePipelineTest.java:47]: Something like:

```
public static String fromResource(String resourceName) {
        try {
            final URL resource = DeclarativePipelineTest.class.getResource(resourceName);
            if (resource == null) {
                throw new NullPointerException("Resource not found for " + resourceName);
            }
            try (Scanner scanner = new Scanner(resource.openStream(), StandardCharsets.UTF_8.name())) {
                return scanner.useDelimiter("\\A").next();
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
}
```

Comment (marek-parfianowicz) [reply src/test/java/hudson/plugins/clover/DeclarativePipelineTest.java:47]: You could even move the entire pipeline script to a separate file. But this is optional.

File: src/test/java/hudson/plugins/clover/CloverWorkflowTest.java (hunk 1 above)
Comment (MarkEWaite) [reply src/test/java/hudson/plugins/clover/CloverWorkflowTest.java]: Done in a7af80737910b2c1f953540011f980c1d700981c

File: src/test/java/hudson/plugins/clover/DeclarativePipelineTest.java (hunk 2 above)
Comment (MarkEWaite) [reply src/test/java/hudson/plugins/clover/DeclarativePipelineTest.java]: Added in 20ab139990542c4fb3153869f6f891302fe125d9

**Review by marek-parfianowicz — APPROVED** [review]

**Comment by MarkEWaite** [comment]
@marek-parfianowicz I've been using this in my Jenkins environment since before it was merged to the master branch.  It is working well.  If you would be willing to release it, that would allow the documentation on https://plugins.jenkins.io/clover/ to match the documentation content in the [README](https://github.com/jenkinsci/clover-plugin#readme).

**Comment by marek-parfianowicz** [comment]
Good point! I will release it this month, in such case.

**Comment by marek-parfianowicz** [comment]
I just released clover 4.13.0. Cheers!

**Comment by MarkEWaite** [comment]
> I just released clover 4.13.0. Cheers!

Thanks very much!

**Pull request merged**